File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/javascript/exercises/practice/robot-name/robot-name.js
// This is only a SKELETON file for the 'Robot Name' exercise. It's been
// provided as a convenience to get your started writing code faster.

export class Robot {
  static usedNames = new Set();

  constructor() {
    this._name = null;
  }

  get name() {
    if (!this._name) {
      let newName;
      do {
        const letters = String.fromCharCode(
          65 + Math.floor(Math.random() * 26),
          65 + Math.floor(Math.random() * 26)
        );
        const digits = Math.floor(Math.random() * 1000).toString().padStart(3, '0');
        newName = letters + digits;
      } while (Robot.usedNames.has(newName));
      
      Robot.usedNames.add(newName);
      this._name = newName;
    }
    return this._name;
  }

  reset() {
    if (this._name) {
      this._name = null;
    }
  }
}

Robot.releaseNames = () => {};

File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/python/exercises/practice/robot-name/robot_name.py
import random
import string

class Robot:
    _used_names = set()

    def __init__(self):
        self._name = None
        self._previous_names = set()

    def reset(self):
        if self._name is not None:
            Robot._used_names.discard(self._name)
            self._previous_names.add(self._name)
            self._name = None

    @property
    def name(self):
        if self._name is None:
            while True:
                letters = ''.join(random.choices(string.ascii_uppercase, k=2))
                numbers = ''.join(random.choices(string.digits, k=3))
                new_name = f"{letters}{numbers}"
                if new_name not in Robot._used_names and new_name not in self._previous_names:
                    Robot._used_names.add(new_name)
                    self._name = new_name
                    break
        return self._name
